android/cameraview/cameraview/src/main/api21/com/google/android/cameraview/Camera2.java

```diff
@@ -170,23 +170,53 @@ public void onReady() {
         @Override
         public void onImageAvailable(ImageReader reader) {
             try (Image image = reader.acquireNextImage()) {
-                Image.Plane[] planes = image.getPlanes();
-                if (planes.length > 0) {
-                    ByteBuffer buffer = planes[0].getBuffer();
+                if (image.getFormat() == ImageFormat.JPEG) {
+                    ByteBuffer buffer = image.getPlanes()[0].getBuffer();
                     byte[] data = new byte[buffer.remaining()];
                     buffer.get(data);
-                    if (image.getFormat() == ImageFormat.JPEG) {
-                        mCallback.onPictureTaken(data);
-                    } else {
-                        mCallback.onFramePreview(data, image.getWidth(), image.getHeight(), mDisplayOrientation);
-                    }
-                    image.close();
```

Comment:
is it safe to remove this?

Reply:
To be honest, I don't remember why I've removed it. I need to re-test having `close()` reverted.

Overall this code works good (we use it in the production app and there are no more issues with the camera/scanner). So seems `image.close()` doesn't lead to memory leaks or other issues.

Reply:
So I get the next hint from the Android Studio when I return `image.close()`.

```
Redundant close less... (⌘F1) 
Inspection reports unnecessary close of resource in the end of try-with-resources block
```

Seems it's safe to remove it.